### PR TITLE
Feature/iteration 2

### PR DESCRIPTION
**Pull request by m1073496**
## What does this PR do?

- Adds a date picker to the DOM
- Updates the DOM with rooms available after selecting (by hitting enter) a date on the date picker
- Adds a dropdown button for filtering results/updates DOM
- Fiercely apologizes
- Adds Book Now buttons to listings
- Makes a POST request after clicking Book Now
- Updates DOM with new booking at the top of the list on user dashboard

## How should it be tested in the UI?

- Overlook my sad little MVP lol, and then just walk through the above^
- There are still some bugs that I know of, but if you find more please let me know!

### CODE REVIEW

File: src/css/_userDashboard.scss [hunk 1]
```diff
@@ -34,7 +34,59 @@ nav {
...
+}
+
+/* Change color of dropdown links on hover */
+.dropdown-content option:hover {background-color: #ddd;}
+
+/* Show the dropdown menu on hover */
+.dropdown:hover .dropdown-content {display: block;}
+
+/* Change the background color of the dropdown button when the dropdown content is shown */
+.dropdown:hover .dropbtn {background-color: #3e8e41;}
```
Comment (monshan) [inline src/css/_userDashboard.scss:89]: Maybe able to nest some of these, but definitely do what's best for your purposes

File: src/index.html [hunk 2]
```diff
@@ -12,11 +13,24 @@ <h1>Overlook</h1>
       <aside class="user-dash">
         <h2>Welcome, <span class="user-greeting"></span></h2>
         <p>Total spent to date: <span class="total-spent"></span></p>
-        <button>Book New Room</button>
+        <button id="book-new-room">Book New Room</button>
+        <label for="start" class="date-picker hidden">Select a date below:</label>
+        <input type="date" id="date" class="date-picker hidden" value="2021-03-10" min="2021-03-10" max="2040-12-31">
```
Comment (monshan) [inline src/index.html:18]: Be wary of these min and max values just considering the dataset is mostly dates for 2020

File: src/index.html [hunk 3]
```diff
@@ -12,11 +13,24 @@ <h1>Overlook</h1>
...
       </aside>
       <main class="bookings">
+        <div class="dropdown hidden">
+          <button id="dropdown-button" class="dropbtn">Room Type</button>
+            <div class="dropdown-content">
+              <option value="Single Room" href="#">Single Room</option>
+              <option value="Suite" href="#">Suite</option>
+              <option value="Junior Suite" href="#">Junior Suite</option>
+              <option value="Residential Suite" href="#">Residential Suite</option>
+            </div>
```
Comment (monshan) [inline src/index.html:28]: If you're interested I believe the semantic tag version of this is <select>!

File: src/index.js [hunk 4]
```diff
@@ -75,3 +185,29 @@ function renderUserDashboard() {
...
+      break;
+    case 'Junior Suite':
+      roomType = 'junior suite';
+      break;
+    case 'Residential Suite':
+      roomType = 'residential suite';
+      break;
+  }
+  filterRooms(roomType);
+})
```
Comment (monshan) [inline src/index.js:213]: Nice use of a switch / case here, though I would raise you that the 'change' or 'input' event might be more suitable for your dropdown event. But this works just as well

File: src/index.js [hunk 5]
```diff
@@ -20,20 +20,34 @@ import Booking from './Booking';
...
+}
 
 function fetchData() {
   Promise.all([allCustomerDataAPI, bookingDataAPI, roomDataAPI])
     .then((values) => {
       createInstances(values);
-      renderUserDashboard()
+      renderUserDashboard();
     })
 }
```
Comment (monshan) [inline src/index.js:53]: Really interested in how you got this to work considering I've been unsuccessful with my global hookups

**Review by monshan — APPROVED** [review]
Overall very nice work for only one day working on the codebase, you're definitely on track

**Comment by m1073496** [comment]
[Iteration 2: Customer Interaction](https://trello.com/c/Q3BscOAj/27-iteration-2-customer-interaction)

**Pull request merged**